scipy/stats/stats.py

```diff
@@ -3490,74 +3498,220 @@ def ttest_ind(a, b, axis=0, equal_var=True):
     >>> stats.ttest_ind(rvs1, rvs5)
     (-1.4679669854490653, 0.14263895620529152)
-    >>> stats.ttest_ind(rvs1, rvs5, equal_var = False)
+    >>> stats.ttest_ind(rvs1, rvs5, equal_var=False)
     (-0.94365973617132992, 0.34744170334794122)
 
+    When performing a permutation test, one usually wants to use a large number
+    of permutations and use a ``RandomState`` instance to ensure complete
+    reproducibility of the result:
+
+    >>> stats.ttest_ind(rvs1, rvs5, permutations=10000,
+    ...                 random_state=np.random.RandomState(12345))
+    (-0.80402445, 0.41685831)
+
     """
     a, b, axis = _chk2_asarray(a, b, axis)
+    if a.size == 0 or b.size == 0:
+        return (np.nan, np.nan)
 
-    Ttest_indResult = namedtuple('Ttest_indResult', ('statistic', 'pvalue'))
+    if permutations is not None:
+        random_state = check_random_state(random_state)
 
-    if a.size == 0 or b.size == 0:
-        return Ttest_indResult(np.nan, np.nan)
+        if a.ndim == 1:
+            a = np.atleast_2d(a).T
+        if b.ndim == 1:
+            b = np.atleast_2d(b).T
 
-    v1 = np.var(a, axis, ddof=1)
-    v2 = np.var(b, axis, ddof=1)
-    n1 = a.shape[axis]
-    n2 = b.shape[axis]
+        if a.shape[1-axis] != b.shape[1-axis]:
+            if a.shape[1-axis] > b.shape[1-axis]:
+                b = np.tile(b, a.shape[1-axis]//b.shape[1-axis])
+            else:
+                a = np.tile(a, b.shape[1-axis]//a.shape[1-axis])
 
-    if equal_var:
-        df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
-    else:
-        df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
+        mat = np.concatenate((a, b), axis=axis)
+        cats = np.hstack((np.zeros(a.shape[axis]), np.ones(b.shape[axis])))
+        res = _permutation_ttest(mat, cats,
+                                 axis=axis,
+                                 equal_var=equal_var,
+                                 permutations=permutations,
+                                 random_state=random_state)
 
-    res = _ttest_ind_from_stats(np.mean(a, axis), np.mean(b, axis), denom, df)
+    else:
+        v1 = np.var(a, axis, ddof=1)
+        v2 = np.var(b, axis, ddof=1)
+        n1 = a.shape[axis]
+        n2 = b.shape[axis]
 
+        if equal_var:
+            df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
+        else:
+            df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
+        res = _ttest_ind_from_stats(np.mean(a, axis),
+                                    np.mean(b, axis),
+                                    denom, df)
+    Ttest_indResult = namedtuple('Ttest_indResult', ('statistic', 'pvalue'))
     return Ttest_indResult(*res)
 
 
+def _init_summation_index(cats):
+    """
+    Creates a matrix filled with category permutations
+
+    cats: numpy.array
+       List of class assignments
+    """
+    c = len(cats)
+    num_cats = len(np.unique(cats))  # Number of distinct categories
+    copy_cats = copy.deepcopy(cats)
```

Comment:
Prefer to have this done with numpy functionality (simply `cats.copy()` should work right?), and then remove the `import copy` at the top

Reply:
yes, this still needs to be changed

Reply:
Done.